test/proposal_test.rb

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "spec_helper"
+require_relative "test_refinements"
+require "storage"
+require "storage/proposal"
+require "storage/fake_probing"
+require "storage/fake_device_factory"
+
+describe Yast::Storage::Proposal do
+  describe "#propose" do
+    using Yast::Storage::TestRefinements
+
+    def input_file_for(name)
+      File.join(DATA_PATH, "input", "#{name}.yml")
+    end
+
+    def output_file_for(name)
+      File.join(DATA_PATH, "output", "#{name}.yml")
+    end
+
+    def fake_scenario(scenario)
+      fake_probing = Yast::Storage::FakeProbing.new
+      devicegraph = fake_probing.devicegraph
+      Yast::Storage::FakeDeviceFactory.load_yaml_file(devicegraph, input_file_for(scenario))
+      fake_probing.to_probed
+    end
```

Comment:
Frankly, I don't like this approach. Previously, the responsibility was distributed: FakeProbing took care about inhibiting the default behaviour, i.e. it prevented the lib from doing real hardware probing, and returned a device graph to be treated as the faked "probed" device graph.

This new fake_scenario() function silently assumes that it is safe to create just any new device graph and then move it to "probed". IMHO this assumption is not safe. This is the current behaviour of libstorage-ng, but this is really an implementation detail that is better shielded by FakeProbing; that's what FakeProbing is for.